From 86d3baa97de4c18ac474725f68586d341b56a184 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:49:47 -0400
Subject: [tools] lib/traceevent: Use pevent_print_func_field in hrtimer_start handler

Message-id: <1408953210-26343-100-git-send-email-jolsa@redhat.com>
Patchwork-id: 88803
O-Subject: [PATCH RHEL7.1 BZ1133083 099/322] tools lib traceevent: Use pevent_print_func_field in hrtimer_start handler
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit b32cea650a16092a654baa2095c78fa917bad4da
Author: Jiri Olsa <jolsa@redhat.com>
Date: Tue Dec 3 14:09:38 2013 +0100

description
===========
The pevent_print_func_field function encompasses all the functionality
used in the hrtimer_start handler. Change the handler to use this
function.

This also unifies the function field output with the
hrtimer_expire_entry handler.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/plugin_hrtimer.c b/tools/lib/traceevent/plugin_hrtimer.c
index e41d4cf..0b0ebf3 100644
--- a/tools/lib/traceevent/plugin_hrtimer.c
+++ b/tools/lib/traceevent/plugin_hrtimer.c
@@ -48,10 +48,6 @@ static int timer_start_handler(struct trace_seq *s,
           struct pevent_record *record,
           struct event_format *event, void *context)
 {
- struct pevent *pevent = event->pevent;
- struct format_field *fn = pevent_find_field(event, "function");
- void *data = record->data;
-
  trace_seq_printf(s, "hrtimer=");
 
  if (pevent_print_num_field(s, "0x%llx", event, "timer",
@@ -59,19 +55,8 @@ static int timer_start_handler(struct trace_seq *s,
   pevent_print_num_field(s, "0x%llx", event, "hrtimer",
            record, 1);
 
- if (!fn) {
-  trace_seq_printf(s, " function=MISSING");
- } else {
-  unsigned long long function;
-  const char *func;
-
-  if (pevent_read_number_field(fn, data, &function))
-   trace_seq_printf(s, " function=INVALID");
-
-  func = pevent_find_function(pevent, function);
-
-  trace_seq_printf(s, " function=%s", func);
- }
+ pevent_print_func_field(s, " function=%s", event, "function",
+    record, 0);
 
  trace_seq_printf(s, " expires=");
  pevent_print_num_field(s, "%llu", event, "expires", record, 1);
-- 
1.7.1